src/System.Text.Json/tests/JsonDocumentTests.cs

```diff
@@ -1834,21 +1842,28 @@ public static void CheckUseDefault()
 
             Assert.Throws<InvalidOperationException>(() =>
             {
-                Utf8JsonWriter writer = default;
-                root.WriteValue(writer);
+                var buffer = new ArrayBufferWriter<byte>(1024);
+                Utf8JsonWriter writer = new Utf8JsonWriter(buffer);                
+                root.WriteTo(writer);
             });
+        }
 
-            Assert.Throws<InvalidOperationException>(() =>
+        [Fact]
+        public static void CheckByPassingNullWriter()
+        {
+            using (JsonDocument doc = JsonDocument.Parse("{\"First\":1}", default))
             {
-                Utf8JsonWriter writer = default;
-                root.WriteProperty(ReadOnlySpan<char>.Empty, writer);
-            });
+                var expectedErrorMessage = @"Value cannot be null. (Parameter 'writer')";
+                AssertExtensions.Throws<ArgumentNullException>(() => doc.WriteTo(null), expectedErrorMessage);
 
-            Assert.Throws<InvalidOperationException>(() =>
-            {
-                Utf8JsonWriter writer = default;
-                root.WriteProperty(ReadOnlySpan<byte>.Empty, writer);
-            });
+                JsonElement root = doc.RootElement;
+                AssertExtensions.Throws<ArgumentNullException>(() => root.WriteTo(null), expectedErrorMessage);
```

Comment:
The tests here on the element and properties belong in JsonElement and JsonProperty -specific tests, they don't need to be repeated here.

Reply:
Done. Removed here and moved into their respective files.